pkg/operator/staticpod/controller/installer/installer_controller_test.go

```diff
@@ -522,19 +518,23 @@ func TestCreateInstallerPodMultiNode(t *testing.T) {
 		}
 	}
 
-	tests := []struct {
-		name                    string
-		nodeStatuses            []operatorv1.NodeStatus
-		staticPods              []*corev1.Pod
-		latestAvailableRevision int32
-		expectedUpgradeOrder    []int
-		expectedSyncError       []bool
-		updateStatusErrors      []error
-		numOfInstallersOOM      int
-	}{
+	type testCase struct {
+		name                 string
+		nodeStatuses         []operatorv1.NodeStatus
+		staticPods           []*corev1.Pod
+		availableRevisions   []int32
+		expectedUpgradeOrder []int
+		expectedSyncError    []bool
+		updateStatusErrors   []error
+		numOfInstallersOOM   int
+		disasterRecoveryOnly bool
+		disasterRecoverySkip bool
+	}
```

Comment:
these are big changes to the tests I am not comfortable with so short before a release. Much simpler: add `ownerRefsFail            map[int32]bool` here and return an error from the fake owner ref setter. That's another 4 liner change, plus a new test case.

Reply:
I was attempting to make minimal changes to the existing tests, and I wanted, for coverage to run them all twice: normally, vs. after the deletion of the `openshift-kube-apiserver` namespace.

I run the test cases twice, once as originally written, and a second time with an environment matching what would be seen after deleting the `openshift-kube-apiserver` (specifically, only a new revision `n+1` available, where `n` is the last revision available before the deletion). In order not to duplicate all the tests with minor changes, I pro-grammatically alter them the second time around to make only the `n+1` revision available and run again.  I did have to duplicate 3 test cases that would result in different results in the two scenarios, hence the need for the flags.

Reply:
I really don't like this complex change 4 days before deadline. This works and covers the issue, doesn't it?

```
                        c.ownerRefsFn = func(revision int32) ([]metav1.OwnerReference, error) {
+                               if test.ownerRefsFail[revision] {
+                                       return nil, fmt.Errorf("ownerRefsFn failed for revision %d", revision)
+                               }
                                return []metav1.OwnerReference{}, nil
                        }
```